remote-index.md

```diff
@@ -1,66 +1,69 @@
 # Remote index
 
+A [project-wide index](/design/indexing.html) can be slow to build, particlarly
+for large projects on slower machines like laptops. A "remote index" allows
+a shared server to host the index instead.
+
 ## How it works
 
-Remote index works in two separate components, an indexing pipeline and a
-serving infrastructure. The former produces monolithic index file for a
-project of interest and the latter serves information from a monolithic index
-file to clangd clients using a [gRPC](https://grpc.io/)-based server.
+The remote index has three components:
+ - the **indexer** is a batch process that should run on a powerful machine.
+   It parses all the code for your project, and produces an index file.
+ - the **server** exposes the index over a network. It loads the index file into
+   memory, and exposes a [gRPC](https://grpc.io) service to query it.
+ - the **client** is part of clangd, which runs on your development machine.
+   It connects to the server and requests information (such as the definition
+   of a symbol) in real-time.
+
+Each of these components are open-source and part of
+[llvm/llvm-project/clang-tools-extra/clangd](https://github.com/llvm/llvm-project/tree/master/clang-tools-extra/clangd/).
 
-## Indexing pipeline
+## Indexer
 
-Indexing pipeline uses `clangd-indexer` under the hood and collects symbols
-that can be exported from headers (functions, classes, types, etc). Running
-`clangd-indexer` is expensive and produced index is not incremental, so using
-the remote index most likely results in a stale index that is periodically
-updated on the server.
+`clangd-indexer` collects symbols that can be exported from headers
+(functions, classes, types, etc).
```

Comment:
Is this supposed to be a separate paragraph or just the end of a sentence? The newline here (way below the usual widthlimit) is a bit misleading.